src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/LSPEditorWrapper.java

```diff
@@ -0,0 +1,246 @@
+                event.getArea() == EditorMouseEventArea.EDITING_AREA &&
+                event.getMouseEvent().getModifiers() == 0 &&
+                event.isOverText() &&
+                event.getCollapsedFoldRegion() == null) {
+            return event.getOffset();
+        }
+        return -1;
+    }
+
+    public Editor getEditor() {
+        return editor;
+    }
+
+    /**
+     * Update Intellij QuickFixes which should be available at the given offset
+     * by collecting LSP code actions at the given offset.
+     *
+     * @param offset the caret / hover offset
+     * @return
+     */
+    private CompletableFuture<Void> updateQuickFixAt(int offset, int start,int end) {
+        Document document = editor.getDocument();
+        VirtualFile file = LSPIJUtils.getFile(document);
+        if (file == null) {
+            return null;
+        }
+        URI fileUri = LSPIJUtils.toUri(document);
+
+        CodeActionParams params = new CodeActionParams();
+        params.setTextDocument(LSPIJUtils.toTextDocumentIdentifier(fileUri));
+        Position pos = LSPIJUtils.toPosition(offset, document);
+        Range range = new Range(pos, pos);
+        params.setRange(range);
+
+        List<Diagnostic> currentDiagnostics = LSPVirtualFileWrapper.getLSPVirtualFileWrapper(file).getAllDiagnostics();
+        List<Diagnostic> diagnosticContext = new ArrayList<>();
+        if (!currentDiagnostics.isEmpty()) {
+            List<Diagnostic> diagnostics = new ArrayList<>(currentDiagnostics);
+            diagnostics.forEach(diagnostic -> {
+                int startOffset = LSPIJUtils.toOffset(diagnostic.getRange().getStart(), document);
+                int endOffset = LSPIJUtils.toOffset(diagnostic.getRange().getEnd(), document);
+                if (offset >= startOffset && offset <= endOffset) {
+                    diagnosticContext.add(diagnostic);
+                }
+            });
+        }
+
+        CodeActionContext context = new CodeActionContext(diagnosticContext);
+        params.setContext(context);
+
+        Project project = LSPIJUtils.getProject(file).getProject();
+        return LanguageServiceAccessor.getInstance(project)
+                .getLanguageServers(document, capabilities -> capabilities.getCodeActionProvider() != null)
+                .thenComposeAsync(languageServers -> CompletableFuture.allOf(languageServers.stream()
+                        .map(languageServer -> languageServer.getTextDocumentService().codeAction(params)
+                                .thenAcceptAsync(codeActions -> {
+                                    // textDocument/codeAction may return null
+                                    if (codeActions != null) {
+                                        this.codeActionResult = new LSPCodeActionResult(context, codeActions, languageServer);
+                                        updateErrorAnnotations(document, project, editor, offset, start, end);

```

Comment:
why not pass `this.codeActionResult` as a parameter to `updateErrorAnnotations`?
I see that the instance var is required elsewhere but I believe that in this particular call it would be better to pass them to the method directly.